src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -431,10 +498,34 @@ protected Query doToQuery(QueryShardContext context) {
             fieldDimension = modelMetadata.getDimension();
             knnEngine = modelMetadata.getKnnEngine();
             spaceType = modelMetadata.getSpaceType();
+            if (modelMetadata.getMethodComponentContext() != null) {
+                method = modelMetadata.getMethodComponentContext().getName();
+            }
         } else if (knnMethodContext != null) {
             // If the dimension is set but the knnMethodContext is not then the field is using the legacy mapping
             knnEngine = knnMethodContext.getKnnEngine();
             spaceType = knnMethodContext.getSpaceType();
+            method = knnMethodContext.getMethodComponentContext().getName();
+        }
+
+        if (!Strings.isNullOrEmpty(method)) {
+            final EngineSpecificMethodContext engineSpecificMethodContext = knnEngine.getMethodContext(method);
+            ValidationException validationException = validateParameters(
+                engineSpecificMethodContext.supportedMethodParameters(),
+                (Map<String, Object>) methodParameters
+            );
+            if (validationException != null) {
+                throw new IllegalArgumentException(
+                    String.format(
+                        "Parameters not valid for [%s]:[%s] combination: [%s]",
+                        knnEngine,
+                        method,
+                        validationException.getMessage()
+                    )
+                );
+            }
+        } else {
+            throw new IllegalStateException("method not found");
```

Comment:
Need confirmation that this is the right behavior and there will never be a scenario which is valid when there is no method found

Reply:
nevermind will have to change it because of an IT